### PR TITLE
add docs for keys and key-sets entities

### PR DESCRIPTION
**Pull request by jschmid1**
### Summary

Add a new reference for `key-management` that describes how `keys` and `key-sets` are used alongside examples.

This is an OSS feature.

### Reason

https://konghq.atlassian.net/browse/DOCU-2786

Things to do before merging:

- [x] the "Reference" section doesn't feel suited. It _is_ a core entity but not as frequently used as routes/services etc. It's probably somewhere alongside "Certificates".
- [x] There is a lot of unformatted/unformulated text in the PR. This needs to be adapted to the style of writing used across this document
- [x] Missing links and references to other sections

### CODE REVIEW

**Comment by cloudjumpercat** [comment]
Hi @jschmid1 ! Just wanted to check to see if these docs are ready for a writer copyedit right now.

**Comment by jschmid1** [comment]
> Hi @jschmid1 ! Just wanted to check to see if these docs are ready for a writer copyedit right now.

@cloudjumpercat yes, should be ready

File: src/gateway/reference/key-management.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
+---
+title: Key and Key Set Management in Kong Gateway
+content_type: reference
+---
+
+Note that this is an OSS feature.
```
Comment (cloudjumpercat) [inline src/gateway/reference/key-management.md:6]: @jschmid1 Does this mean it's OSS only, or that it's a free feature and anyone can use it regardless of if they are OSS, plus, or enterprise?

**Review by cloudjumpercat — COMMENTED** [review]
Thanks for submitting this! I left a comment for you on the PR and made a few copyedits.

File: src/gateway/reference/key-management.md (hunk 1 above)
Comment (jschmid1) [reply src/gateway/reference/key-management.md:6]: > or that it's a free feature and anyone can use it regardless of if they are OSS, plus, or enterprise?

this ^^^

File: src/gateway/reference/key-management.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
```
Comment (cloudjumpercat) [inline src/gateway/reference/key-management.md:6]:
```suggestion
content_type: reference
badge: free
---
```

**Review by cloudjumpercat — APPROVED** [review]
LGTM!

**Pull request merged**